### PR TITLE
Add code coverage for GitHub Actions

### PR DESCRIPTION
**Pull request by oliverklee**

### CODE REVIEW

**Comment by oliverklee** [comment]
This needs the following environment variables in the GitHub repository configuration to run:

```
- GITHUB_REF
- GITHUB_ACTIONS
- GITHUB_RUN_ID
- GITHUB_EVENT_NAME
- COVERALLS_REPO_TOKEN
```

**Comment by Danon** [comment]
@oliverklee Is this still a draft?

**Comment by oliverklee** [comment]
> Is this still a draft?

Yes, I'll need to rebase. Hold on …

**Comment by oliverklee** [comment]
This now needs the corresponding configuration variables set to succeed. @Danon Would you be willing to take care of this?

**Comment by Danon** [comment]
@oliverklee Sure, but would you be willing to explain what should they contain exactly? I'm new to github actions.

```
- GITHUB_REF
- GITHUB_ACTIONS
- GITHUB_RUN_ID
- GITHUB_EVENT_NAME
```

PS: Is name `GITHUB_REF` a real name? Or did you come up with it? If it's the second, is it short for `GITHUB_REFERENCE` or anything like that?

**Comment by oliverklee** [comment]
I'm not sure - this is the list output by Coveralls, and I'm not familiar with Coveralls (yet).

We might also want to use a dedicated GitHub Action for this. At least the discussion at https://stackoverflow.com/questions/60362121/how-to-configure-coveralls-with-github-action points in that direction.

**Comment by oliverklee** [comment]
This did not work. I'm going to try something else now.

**Comment by Danon** [comment]
> o try something else now.

Ok waiting

**Comment by oliverklee** [comment]
@Danon Which environment variables do you have for Coveralls in your Travis CI configuration?

**Comment by Danon** [comment]
@oliverklee I don't think there are any environment variables.

**Comment by oliverklee** [comment]
Then how does the authentication for pushing to Coveralls work?

**Comment by Danon** [comment]
> Then how does the authentication for pushing to Coveralls work?

If we're using php-coveralls, perhaps we can take a look at this instruction here: https://github.com/php-coveralls/php-coveralls#github-actions

**Comment by oliverklee** [comment]
I think I need your help here. I have never worked with Coveralls before and don't know my way around it.

**Comment by Danon** [comment]
> I think I need your help here. I have never worked with Coveralls before and don't know my way around it.

In the `php-coveralls` read me, we read that to set it up on github actions it would only require to add this step

> GitHub Actions
> Add a new step after phpunit generate coverage report.
> ```yaml
> - name: Upload coverage results to Coveralls
>   env:
>     COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
>   run: |
>     composer global require php-coveralls/php-coveralls
>     php-coveralls --coverage_clover=build/logs/clover.xml -v
> ```

And `secrets.GITHUB_TOKEN` is already added.

**Comment by oliverklee** [comment]
It worked! I'll nicely squash the commits now and repush.

**Comment by oliverklee** [comment]
Done.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         with:
           php-version: ${{ matrix.php-version }}
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (Danon) [inline .github/workflows/ci.yml:34]: @oliverklee Are you sure this is needed for CI build?

Comment (oliverklee) [reply .github/workflows/ci.yml:34]: I can try if it works without it, but I doubt it. Hold on.

**Comment by oliverklee** [comment]
No, doesn't work. Reverting.

**Pull request merged**